server/src/main/java/io/druid/server/coordinator/rules/BroadcastDistributionRule.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.coordinator.rules;
+
+import com.metamx.emitter.EmittingLogger;
+import io.druid.server.coordinator.CoordinatorStats;
+import io.druid.server.coordinator.DruidCoordinator;
+import io.druid.server.coordinator.DruidCoordinatorRuntimeParams;
+import io.druid.server.coordinator.ServerHolder;
+import io.druid.timeline.DataSegment;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public abstract class BroadcastDistributionRule implements Rule
+{
+  private static final EmittingLogger log = new EmittingLogger(BroadcastDistributionRule.class);
+
+  @Override
+  public CoordinatorStats run(
+      DruidCoordinator coordinator, DruidCoordinatorRuntimeParams params, DataSegment segment
+  )
+  {
+    // Find servers which holds the segments of co-located data source
+    final List<ServerHolder> targetServerHolders = params
+        .getDruidCluster().getAllServers().stream()
+        .filter(eachHolder -> eachHolder.getServer().getDataSource(getColocatedDatasource()) != null)
```

Comment:
It seems to me like it'd be valuable to have `colocatedDataSources` be a list, and maybe if it's empty then broadcast to _all_ servers. That'd give some more flexibilty to the user about where to broadcast their data sources.